packages/react/src/components/navigation/FlowStepper/index.tsx

```diff
@@ -42,11 +47,19 @@ export interface Props<ExtraProps> {
   /**
    * An optional header displayed above the stepper.
    */
-  header?: (props: InnerProps & ExtraProps) => React.ReactNode;
+  header?: ((props: InnerProps & ExtraProps) => React.ReactNode) | StepExtra[];
```

Comment:
i dont really like those mixed types especially when one is an array and the other not :thinking: 
it forces you to check for types all along the component.
i would suggest in those cases splitting this field in 2 seperate ones 
```js
header?: ((props: InnerProps & ExtraProps) => React.ReactNode);
stepHeaders?: StepExtra[];

...

const headerContents = header || stepHeaders.find(...);
...
```

Reply:
or only use a list and check on length :shrug:  to not have to mix types